### ISSUE TITLE
Bug: return values that is a union is dependent on order

### ISSUE THREAD

**Issue by einarwar** [jsh9/pydoclint#84]
Thanks for starting this project. As a current user of darglint, i am trying to migrate to using pydoclint and oh boy is it much better!

I noticed a unexpected error. The following code will raise a pydoclint violation:

```python
from typing import Any

def test_function() -> dict[str, Any] | None:
    """Some function

    Returns:
        dict[str, Any] | None: Something
    """
    ...
```
```
DOC203: Function `test_function` return type(s) in docstring not consistent with the return annotation. Return annotation types: ['dict[str, Any] | None']; docstring return section types: ['']
```

However, flipping the union around does not raise any violations
```python
def test_function_2() -> None | dict[str, Any]:
    """Some other function

    Returns:
        None | dict[str, Any]: Something else
    """
    ...

# No errors here
```

As far as i know, the Union operator is commutative, i.e `X | Y == Y | X`, so neither of these should raise errors

**Comment by jsh9** [comment]
Hi @einarwar , thanks for posting this issue.

This may be a bug in parsing the docstring rather than in the order.

I'm quite busy this week, so I may not have time to look into it in detail until next week.

**Comment by jsh9** [comment]
It's indeed an issue with the docstring parser.  I'm looking for ways to fix it now.  It may take a few days.

**Comment by jsh9** [comment]
The bug is in the underlying dependency (https://github.com/rr-/docstring_parser).  Fortunately, I forked it a few weeks ago, so that I can quickly make bug fixes.  This is the underlying bug fix, in case you are curious: https://github.com/jsh9/docstring_parser_fork/pull/5

Then in pydoclint, all I needed was to update dependency.

**Comment by jsh9** [comment]
Hi @einarwar , I fixed this issue.

Please update to version [0.3.6](https://github.com/jsh9/pydoclint/releases/tag/0.3.6) on PyPI.

**Comment by einarwar** [comment]
Thanks for solving this quickly. My team is very happy for migrating from darglint to pydoclint :)

**Comment by jsh9** [comment]
> Thanks for solving this quickly. My team is very happy for migrating from darglint to pydoclint :)

Thank you!  If you have any more feedback, please feel free to open new issues/discussions.
